### PR TITLE
Fixes #3359

### PR DESCRIPTION
**Pull request by siddhant2002**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.

### CODE REVIEW

**Review by siriak — COMMENTED** [review]
Which algorithm is that?

**Comment by siriak** [comment]
image
File names are not in PascalCase

**Comment by siddhant2002** [comment]
> Which algorithm is that?

It displays all the paths from source to target

**Comment by siddhant2002** [comment]
@siriak please check the updated pr

**Comment by siddhant2002** [comment]
@siriak  please check the updated pr

File: src/main/java/com/thealgorithms/backtracking/AllPathsFromSourceToTarget.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        }
+
+        // Mark the current node
+        isVisited[u] = false;
+        return a[0];
+        // returns number of paths from source to destination
+    }
+
+    // Driver program
+    public static boolean all_Paths_From_Source_To_Target(int vertices, int a[][], int source, int destination, int num_of_paths)
```
Comment (siriak) [inline src/main/java/com/thealgorithms/backtracking/AllPathsFromSourceToTarget.java:105]: Please make it return all paths as `int[][]` and rename it to camelCase

File: src/main/java/com/thealgorithms/backtracking/AllPathsFromSourceToTarget.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        // initialise vertex count
+        this.v = vertices;
+
+        // initialise adjacency list
+        initAdjList();
+    }
+
+    // utility method to initialise adjacency list
+    
+    @SuppressWarnings("unchecked")
```
Comment (siriak) [inline src/main/java/com/thealgorithms/backtracking/AllPathsFromSourceToTarget.java:37]: Is this necessary? Please fix the warning without suppressing it

**Review by siriak — CHANGES_REQUESTED** [review]

File: src/main/java/com/thealgorithms/backtracking/AllPathsFromSourceToTarget.java (hunk 1 above)
Comment (siddhant2002) [reply src/main/java/com/thealgorithms/backtracking/AllPathsFromSourceToTarget.java]: > Please make it return all paths as `int[][]` and rename it to camelCase

I've changed it to camel case

Comment (siddhant2002) [reply src/main/java/com/thealgorithms/backtracking/AllPathsFromSourceToTarget.java]: > Please make it return all paths as `int[][]` and rename it to camelCase
I've passed the answer array to the method and if the answer array matches it will return true. So, I've kept the method return type as boolean.

File: src/main/java/com/thealgorithms/backtracking/AllPathsFromSourceToTarget.java (hunk 2 above)
Comment (siddhant2002) [reply src/main/java/com/thealgorithms/backtracking/AllPathsFromSourceToTarget.java]: > Is this necessary? Please fix the warning without suppressing it

No, it's not necessary

**Comment by siddhant2002** [comment]
@siriak I've done the necessary changes. Please check the pr. The return type cannot be changed to int [ ][ ] as the junit are being used.

File: src/main/java/com/thealgorithms/backtracking/AllPathsFromSourceToTarget.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+/** Program description - To find all possible paths from source to destination*/
+
+/**Wikipedia link -> https://en.wikipedia.org/wiki/Shortest_path_problem */
+package com.thealgorithms.backtracking;
+
+// JAVA program to print all paths from a source to destination.
+import java.util.*;
+
+// A directed graph using adjacency list representation
+
```
Comment (siriak) [inline src/main/java/com/thealgorithms/backtracking/AllPathsFromSourceToTarget.java:14]: What is this comment related to?

File: src/main/java/com/thealgorithms/backtracking/AllPathsFromSourceToTarget.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
+/** Author : Siddhant Swarup Mallick
+ * Github : https://github.com/siddhant2002
+ */
+
+/** Program description - To find all possible paths from source to destination*/
+
+/**Wikipedia link -> https://en.wikipedia.org/wiki/Shortest_path_problem */
+package com.thealgorithms.backtracking;
+
+// JAVA program to print all paths from a source to destination.
```
Comment (siriak) [inline src/main/java/com/thealgorithms/backtracking/AllPathsFromSourceToTarget.java:10]: No need to state that it's in Java, that's already clear

File: src/main/java/com/thealgorithms/backtracking/AllPathsFromSourceToTarget.java [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+        }
+
+        // Mark the current node
+        isVisited[u] = false;
+        return a[0];
+        // returns number of paths from source to destination
+    }
+
+    // Driver program
+    public static boolean allPathsFromSourceToTarget(int vertices, int a[][], int source, int destination, int num_of_paths)
```
Comment (siriak) [inline src/main/java/com/thealgorithms/backtracking/AllPathsFromSourceToTarget.java:103]: How can I get all paths with this function?

Comment (siriak) [inline src/main/java/com/thealgorithms/backtracking/AllPathsFromSourceToTarget.java:103]:
```suggestion
    public static boolean allPathsFromSourceToTarget(int vertices, int a[][], int source, int destination, int num_of_paths)
```
Parameter names are not camelCase

File: src/main/java/com/thealgorithms/backtracking/AllPathsFromSourceToTarget.java [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+        int a[]=new int[1];
+        // Call recursive utility
+        printAllPathsUtil(s, d, isVisited, pathList,a);
+        return a[0];
+    }
+
+    // A recursive function to print all paths from 'u' to 'd'.
+    // isVisited[] keeps track of vertices in current path.
+    // localPathList<> stores actual vertices in the current path 
+    private int printAllPathsUtil(Integer u, Integer d, boolean[] isVisited, List<Integer> localPathList, int a[])
```
Comment (siriak) [inline src/main/java/com/thealgorithms/backtracking/AllPathsFromSourceToTarget.java:70]: Please remove functions related to printing, as this must be a responsibility of a different class. This one is concerned with finding the paths

**Review by siriak — CHANGES_REQUESTED** [review]

**Review by siriak — CHANGES_REQUESTED** [review]
The code looks good, please fix failing tests
Error:    AllPathsFromSourceToTargetTest.testForFirstCase:18 iterable contents differ at index [0][0], expected: <0> but was: <2>
Error:    AllPathsFromSourceToTargetTest.testForFourthcase:51 iterable contents differ at index [0][2], expected: <3> but was: <2>
Error:    AllPathsFromSourceToTargetTest.testForThirdCase:40 iterable contents differ at index [0][0], expected: <0> but was: <1>

**Review by siriak — APPROVED** [review]
Looks good, thanks!
